tools/BatchModelGenerator/Program.cs

```diff
@@ -392,13 +399,20 @@ private static void GenerateGenericCollectionProperty(PropertyInfo property, str
                 CodePropertyReferenceExpression enumeratorCurrent = new(enumeratorReference, "Current");
 
                 // Fill the list by individually wrapping each item in the loop
-                if (magicalDictConversion)
+                if (magicalDictConversion && !regularDictoary)
                 {
                     var keyReference = new CodePropertyReferenceExpression(enumeratorCurrent, OMToPSDictionaryConversionMappings[argType.FullName].Item1);
                     var valueReference = new CodePropertyReferenceExpression(enumeratorCurrent, OMToPSDictionaryConversionMappings[argType.FullName].Item2);
                     CodeMethodInvokeExpression addToList = new(reference, "Add", keyReference, valueReference);
                     loopStatement.Statements.Add(addToList);
                 }
+                else if (regularDictoary) // special case to handel pure Dictionary<string,string> type
```

Comment:
What's this file for?

Reply:
Batch has custom code generation on top of the default powershell generation which needed an update to handle the newest batch C# dataplane sdk.   In the most recent batch C# dataplane sdk we introduced a property of type Dictionary<string,string> in which we have never had such a type before.